### ISSUE TITLE
Enabling textprop will break Sign; setting LspErrorHighlight has no effect

### ISSUE THREAD

**Issue by ApolloTang** [prabirshrestha/vim-lsp#399]
I am trying out vim-lsp, everything works great! Just wanna report some bug. 
and thanks for creating this plugin !!! 

here is my minimum vimrc (which can be found [here](https://github.com/ApolloTang/vim-sandbox/tree/master/0100-installing-lsp/02-lsp-minimum-w-config)):
```
set nocompatible
    set hidden          " Handle multiple buffers better.
    set laststatus=2    " Show the status line all the time

    let DIR = getcwd()
    echom DIR

    if !filereadable(DIR.'/dot-vim/plug.vim')
      echom 'executing curl...'
      silent !curl --insecure -fLo ./dot-vim/plug.vim
        \ https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
    endif

    call plug#begin(DIR.'/dot-vim/plugged')
      "/ vim-lsp
      Plug 'prabirshrestha/async.vim'
      Plug 'prabirshrestha/vim-lsp'

      "/ Typescript syntax
      Plug 'https://github.com/leafgarland/typescript-vim'

      "/ Theme
      Plug 'https://github.com/morhetz/gruvbox'
      Plug 'https://github.com/ApolloTang/apollotang-light'
    call plug#end()


    syntax enable     " Turn on syntax highlighting.
    if has("gui_running")
      if has("gui_macvim")
        echom 'is macvim'
        " colorscheme gruvbox
        colorscheme apollotang-light
        set guifont=Inconsolata:h14
      endif
    endif


    "/ vim-lsp {
      let g:lsp_log_verbose = 1
      let g:lsp_log_file = 'vim-lsp.log'

      if executable('typescript-language-server')
        echom 'typescript-language-server is executable'
        au User lsp_setup call lsp#register_server({
          \ 'name': 'typescript support using typescript-language-server',
          \ 'cmd': {server_info->[&shell, &shellcmdflag, 'typescript-language-server --stdio']},
          \ 'root_uri':{server_info->lsp#utils#path_to_uri(lsp#utils#find_nearest_parent_file_directory(lsp#utils#get_buffer_path(), 'tsconfig.json'))},
          \ 'whitelist': ['typescript', 'typescript.tsx'],
          \ })
      else
        echohl ErrorMsg
        echom 'Sorry, `typescript-language-server` is not installed. See :h vim-lsp-typescript for more details on setup.'
        echohl NONE
      endif


      "/ Diagnostics
      "/ ------------
        let g:lsp_diagnostics_enabled = 1     "/ enable diagnostics
        let g:lsp_diagnostics_echo_cursor = 1 "/ enable echo under cursor when in normal mode


      "/ Sign
      "/ ------------
        let g:lsp_signs_enabled = 1                 "/ Enable signs
        let g:lsp_signs_error = {'text': '✗'}
        let g:lsp_signs_warning = {'text': '‼'}
        let g:lsp_signs_hint = {'text': '*'}


      "/ Virtual text
      "/ ------------
        let g:lsp_virtual_text_enabled = 0          "/ NeoVim only so disable


      "/ Highlight
      "/ ---------
        let g:lsp_highlights_enabled = 1          "/ Enable highlight
        let g:lsp_textprop_enabled = 0            "/ NOTE: Enabling textprop will break Sign



        "/ To change the style of the highlighting, you can set or link:
        "/ LspErrorHighlight`, `LspWarningHighlight`, `LspInformationHighlight` and LspHintHighlight` highlight groups

        "/ *****
        "/ ***** THESE SETTING HAS NO EFFECT
        "/ ***** Expecting to highlight where error occurs but seem to be broken
        "/ *****

          highlight LspErrorHighlight term=underline cterm=underline gui=underline
          "highlight LspErrorHighlight ctermfg=red guifg=red ctermbg=green guibg=green
    "/ }
```

I noted two bugs: 

**One**: 
when: 
```
let g:lsp_textprop_enabled = 1
```
sign will disappear:

before ( `let g:lsp_textprop_enabled = 0`):
![Screen Shot 2019-06-09 at 7 45 00 PM](https://user-images.githubusercontent.com/1150961/59165773-24c1ac80-8aef-11e9-96ca-a1cf017ce6bc.png)
after ( `let g:lsp_textprop_enabled = 1`):
![Screen Shot 2019-06-09 at 7 42 05 PM](https://user-images.githubusercontent.com/1150961/59165779-3efb8a80-8aef-11e9-98fc-06a7f1e12578.png)

To be honest I don't really know what lsp_textprop intended to do, but it shouldn't be turning off sign, am i correct ? 

**Two**:
I am not too sure the effect of `LspErrorHighlight` `LspWarningHighlight`, `LspInformationHighlight` and `LspHintHighlight` but I assume they are used to style the tokens in question -- but I can't seen to produce any effect. And I do like to style the location of error just like YouCompleteMe has been able to do.

I suspected these two bugs are related ?? 

log file is available [here](https://github.com/ApolloTang/vim-sandbox/blob/master/0100-installing-lsp/02-lsp-minimum-w-config/vim-lsp-w-textprop-disable_log.txt)

My vim version:
![Screen Shot 2019-06-09 at 7 37 29 PM](https://user-images.githubusercontent.com/1150961/59165721-1f179700-8aee-11e9-8dcb-003309f1c18a.png)

**Comment by tsufeki** [comment]
This seems to be vim bug vim/vim@0a2f578e22de7e4d82075578afdd5fc2d2dd8134 - because `prop_remove` crashes, the rest of textprop and signs code is not called.

This can be fixed by either requiring patch 8.1.1035 or changing calls to `prop_remove({ ... }, 1, 999999)` I guess?

**Comment by ApolloTang** [comment]
macVim has been updated https://github.com/macvim-dev/macvim/releases/tag/snapshot-156
close! 
thanks
